### DISCUSSION TITLE
Class B example- NucleoL073

### DISCUSSION THREAD

**Discussion by jariy** [Lora-net/LoRaMac-node#1070] [Q&A] [answered]
Where could I find a Class B example? According to this [link](https://github.com/Lora-net/LoRaMac-node/issues/140), there was a example, but I cannot find it.

**Comment by mluis1** [comment, marked as answer]
In recent versions of this project we decided to have a single example supporting all LoRaWAN classes.

The `periodic-uplink-lpp` example supports all 3 classes.

By default the provided example is setup to use ClassA.
In order to use ClassB one must modify the below line on `main.c` file.

https://github.com/Lora-net/LoRaMac-node/blob/9a6c7dc02e01282c0d6d9102196abe7925e81a49/src/apps/LoRaMac/periodic-uplink-lpp/NucleoL073/main.c#L48-L51

Please change `CLASS_A` into `CLASS_B`

Something you also need to change is the uplink transmission periodicity in order to have the chance to open a few ping-slots in between uplinks.

https://github.com/Lora-net/LoRaMac-node/blob/9a6c7dc02e01282c0d6d9102196abe7925e81a49/src/apps/LoRaMac/periodic-uplink-lpp/NucleoL073/main.c#L53-L62

As first test I would change the values to from `5000` and `1000` to `50000` and `10000` respectively.

The expected end-device behavior is as follows:
- Start as ClassA device
- Start acquiring the ClassB beacon
- Once beacon is acquired inform the network server about the wanted ping-slot periodicity.
- Once the network server answers to ping-slot periodicity MAC command the end-device is able to switch to ClassB
- From now on the network server will be able to schedule downlinks on ping-slots reception window.

**Reply by jariy** [reply]
Thanks, I changed it to 120000 and 10000 to test the beacon window expansions

**Reply by mluis1** [reply]
Nice. if you don't mind I mark this topic as answered.
